### PR TITLE
Section keyword 'SUMMARY' was incorrectly identified as UDQ a keyword

### PR DESCRIPTION
**Pull request by joakim-hove**
Thanks to @hnil  and @bska for spotting and explaining the issue.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp [hunk 1]
```diff
@@ -576,8 +576,14 @@ SummaryConfig::SummaryConfig( const Deck& deck,
                               ErrorGuard& errors,
                               const GridDims& dims) {
     SUMMARYSection section( deck );
-    for( auto& x : section )
-        handleKW( this->keywords, x, schedule, tables, parseContext, errors, dims);
+    auto kw_iter = section.begin();
+    if (kw_iter != section.end())
+        kw_iter++;
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp]: Maybe add a comment here that this is to skip `SUMMARY`?

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp]: done

Comment (bska) [reply src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp]: Looks good.  Thanks!

**Comment by joakim-hove** [comment]
jenkins build this please

File: tests/parser/SummaryConfigTests.cpp [hunk 2]
```diff
@@ -148,6 +148,14 @@ BOOST_AUTO_TEST_CASE(wells_all) {
             names.begin(), names.end() );
 }
 
+BOOST_AUTO_TEST_CASE(EMPTY) {
+    auto deck = createDeck_no_wells( "" );
+    EclipseState state( deck );
+    Schedule schedule(deck, state.getInputGrid(), state.get3DProperties(), state.runspec());
+    SummaryConfig conf(deck, schedule, state.getTableManager());
+    BOOST_CHECK_EQUAL( conf.size(), 0 );
```
Comment (bska) [inline tests/parser/SummaryConfigTests.cpp]: This check generates a build failure:
```
tests/parser/SummaryConfigTests.cpp:168:29: error: 'class Opm::SummaryConfig' has no member named 'size'
     BOOST_CHECK_EQUAL( conf.size(), 0 );
```

**Comment by joakim-hove** [comment]
Another attempt ...

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**